pages/_app.js

```diff
@@ -1,53 +1,81 @@
 
+import { useMemoryStatus } from '../utils/hooks';
 import AnimationEmulationContext from '../components/AnimationEmulationContext';
 
-class MyApp extends App {
-  state = {
-    manualEnabled: false,
-    isAnimationOn: true
-  };
-
-  componentDidMount = () => {
-    const manualEnabled = localStorage.getItem('manual-enabled');
-    const isAnimationOn = localStorage.getItem('is-animation-on');
+const Loading = () => <Fragment>Loading...</Fragment>;
 
+const MyApp = ({ Component, pageProps, router }) => {
+  const [manualEnabled, setManualEnabled] = useState(false);
+  const [isAnimationOn, setIsAnimationOn] = useState(true);
+  useEffect(() => {
     if (manualEnabled) {
-      this.setState({manualEnabled});
+      setManualEnabled(manualEnabled);
     }
 
     if (isAnimationOn) {
-      this.setState({isAnimationOn});
+      setIsAnimationOn(isAnimationOn);
     }
-  };
+  }, []);
 
-  enableManualAnimationHandler = flag => {
-    localStorage.setItem('manual-enabled', flag);
-    this.setState({manualEnabled: flag});
+  const memoryStatus = useMemoryStatus();
+  if (!memoryStatus) return <Loading />;
+  const { overLoaded } = memoryStatus;
+
+  let animationEnabled = true;
+  if (manualEnabled) {
+    animationEnabled = isAnimationOn;
+  } else {
+    animationEnabled = !overLoaded;
+  }
+
+  const enableManualAnimationHandler = flag => {
+    setManualEnabled(flag);
   };
 
-  toggleAnimationHandler = event => {
-    localStorage.setItem('is-animation-on', event.target.checked);
-    this.setState({isAnimationOn: event.target.checked});
+  const toggleAnimationHandler = event => {
+    setIsAnimationOn(event.target.checked);
   };
 
-  render() {
-    const { Component, pageProps, router } = this.props;
-    const { manualEnabled, isAnimationOn } = this.state;
+  if (animationEnabled) {
     return (
       <AnimationEmulationContext.Provider
         value={{
           manualEnabled,
           isAnimationOn,
-          enableManualAnimationHandler: this.enableManualAnimationHandler,
-          toggleAnimationHandler: this.toggleAnimationHandler
+          animationEnabled,
+          enableManualAnimationHandler: enableManualAnimationHandler,
+          toggleAnimationHandler: toggleAnimationHandler
         }}>
+        <AnimatePresence exitBeforeEnter>
           <Component {...pageProps} key={router.route} />
+        </AnimatePresence>
       </AnimationEmulationContext.Provider>
     );
+  } else {
+    return (
+      <AnimationEmulationContext.Provider
+        value={{
+          manualEnabled,
+          isAnimationOn,
+          animationEnabled,
+          enableManualAnimationHandler: enableManualAnimationHandler,
+          toggleAnimationHandler: toggleAnimationHandler
+        }}>
+        <Component {...pageProps} key={router.route} />
+      </AnimationEmulationContext.Provider>
+    );
+  }
+}
+
+MyApp.getInitialProps = async ({Component, ctx}) => {
+  let pageProps = {};
```

Comment:
Please do not define data type like {}
`let pageProps;` is enough so later we can just check `pageProps` like `if (pageProps)`

Reply:
defined again.